pallets/dapp-staking-v3/src/lib.rs

```diff
@@ -1381,14 +1390,26 @@ pub mod pallet {
                 Error::<T>::RewardExpired
             );
 
-            let (amount, tier_id) =
+            let (mut amount, ranked_tier) =
                 dapp_tiers
                     .try_claim(dapp_info.id)
                     .map_err(|error| match error {
                         DAppTierError::NoDAppInTiers => Error::<T>::NoClaimableRewards,
                         _ => Error::<T>::InternalClaimDAppError,
                     })?;
 
+            let (tier_id, rank) = ranked_tier.deconstruct();
+
+            // if dapp is ranked and there's reward per rank then include the extra reward to the total amount
+            if T::RankingEnabled::get() && !rank.is_zero() {
+                if let Some(reward_per_rank) =
+                    RankRewards::<T>::get(&era).get(tier_id as usize).copied()
+                {
+                    let additional_reward = reward_per_rank.saturating_mul(rank.into());
+                    amount = amount.saturating_add(additional_reward);
+                }
+            }
```

Comment:
I understand we avoid migration with this approach, but it looks weird that post-processing of the amount is required. `try_claim` should be able to handle everything.

A simple migration should do it, e.g. using the `pallet-dapp-staking-migration` to handle multi-stage migration in case we still don't have access to it from polkadot-sdk.

Also keeps the number of reads for reward payout same as before, not increasing the tx fee (I know, minor thing but still).

Reply:
I was trying to avoid migration. Not just for us but also for front-end team. Since reward coming from rank is additional, I thought to separate them. I can do it the way you suggest, it will just take more work

Reply:
I'd suggest to do the change.

For the frontend team, they'll need to adjust it anyways, since they do reward estimation.

Reply:
Btw, if you disagree, that's fine.

Let's proceed with this as it is, and make a future improvement item to optimize storage scheme. We've already identified some redundant fields/items so we could make it all a part of a single migration to save some time.

Reply:
I thought they use onchain data. If you have to rework then will do the migration.

Reply:
They use on-chain data to estimate the rewards before users calls `claim`. Best to see with them though.